### PR TITLE
Use image widget in proofreading interfaces

### PR DESCRIPTION
**Pull request by 70ray**
Both standard and enhanced.

Sandbox at: https://www.pgdp.org/~rp31/c.branch/enh_proof_widget

### CODE REVIEW

**Review by srjfoo — COMMENTED** [review]
I've taken a look in both proofreading interfaces, moved the control bar around, played with the zoom a bit, and checked out WC and Preview. It all looks good!

Is there anything else I should be looking at?

File: faq/prooffacehelp.php
Comment (cpeel) [reply faq/prooffacehelp.php:494]: I would defer updating `prooffacehelp`. I'm dubious how useful that is anyway and it's unclear how we would offer links to it within the Image Viewer Widget.

We do need to pull out the image zoom from userprefs, the initial DB schema, and a DB migration script as part of this. I can help with that if we want to do merge this into a temporary branch instead of `master`.

**Comment by cpeel** [comment]
In the enhanced interface, the Image Widget doesn't seem to take up the full pane at some sizes. This is on Chrome on MacOS.
Screen Shot 2021-07-29 at 4 30 07 PM

**Comment by srjfoo** [comment]
> In the enhanced interface, the Image Widget doesn't seem to take up the full pane at some sizes. This is on Chrome on MacOS.
> Screen Shot 2021-07-29 at 4 30 07 PM

For what it's worth, Firefox on macOS: 
image

The text pane has the same problem in the Enhanced proofing interface, though neither does in WC: 
image

Maybe the problem isn't with the image pane, but with the way the text block is displayed within the text page?

File: faq/prooffacehelp.php
Comment (srjfoo) [reply faq/prooffacehelp.php:494]: > I would defer updating `prooffacehelp`. I'm dubious how useful that is anyway and it's unclear how we would offer links to it within the Image Viewer Widget.

I think that the image-related stuff that will no longer be available does need to be pulled out, because those buttons/links won't be in the interfaces anymore. Other than that, I agree -- I think at a minimum, we can wait.

Even if we do decide to update `prooffacehelp` to describe the Image Viewer Widget, though, I don't think we'd need to provide links to it within the widget. As far as I know, the only place on a proofing page that links to it is the `Help` link in the documentation section of the Toolbox. All that would be needed is updating the descriptions of the functionality. But I agree that I don't know how useful it would be. All of the buttons in the widget have tooltips, and are pretty self-explanatory.

Comment (cpeel) [reply faq/prooffacehelp.php:494]: I was unclear, sorry about that. I agree we should remove references to the old controls but do not think we should add docs on new controls.

File: tools/proofers/image_block_enh.inc
Comment (chrismiceli) [inline tools/proofers/image_block_enh.inc:41]: maybe just remove if we don't want it.

File: tools/proofers/image_frame_std.php
Comment (chrismiceli) [inline tools/proofers/image_frame_std.php:35]: i18n

File: tools/proofers/proof_frame_enh.inc
Comment (chrismiceli) [inline tools/proofers/proof_frame_enh.inc:46]: already present, but could probably use a more informative name :).

File: faq/prooffacehelp.php
Comment (70ray) [reply faq/prooffacehelp.php:494]: > `Set Image Zoom Percent` needs to go, too. -- that icon and function no longer exist where they are.
> 
I've removed this and also the texts that go with the removed lines.

> I believe that the corresponding image manipulation function for the standard interface needs to come out, too (approximagely lines 426-438).
> 
Done this also.

> `<tb>`
> 
> Related, but not in this MR -- should the image zoom come out of the userprefs, or is that still used to set initial zoom settings for initial sessions in a new project? A really quick test (not exhaustive) seems to indicate that it goes by the existing local setting, not by what's in the userprefs, so should probably come out of the userprefs?

The setting is not used in the image widget but it is still used in md_phase2.php. I don't know what to do about it. Possible options:
1. enable metadata temporarily and consider whether to use the image widget in md_phase2.php.
2. remove all the metadata code.
3. something else.

**Comment by 70ray** [comment]
> In the enhanced interface, the Image Widget doesn't seem to take up the full pane at some sizes. This is on Chrome on MacOS.
> Screen Shot 2021-07-29 at 4 30 07 PM

The proportions of the panes are similar in the old interface but it looks odd here. I've added a border to the pane which makes it look more uniform with the text pane.

File: tools/proofers/image_block_enh.inc
Comment (70ray) [reply tools/proofers/image_block_enh.inc:41]: Done this.

File: tools/proofers/image_frame_std.php
Comment (70ray) [reply tools/proofers/image_frame_std.php:35]: OK in principle but this is in a frame so won't ever be seen and we want to replace it all anyway soon.

File: tools/proofers/proof_frame_enh.inc
Comment (70ray) [reply tools/proofers/proof_frame_enh.inc:46]: Good idea. But unless we dig into it to discover exactly what it does any new name could be misleading.

Comment (chrismiceli) [reply tools/proofers/proof_frame_enh.inc:46]: absolutely

File: tools/proofers/image_frame_std.php
Comment (chrismiceli) [reply tools/proofers/image_frame_std.php:35]: Expected as such. Thanks

**Review by chrismiceli — APPROVED** [review]

File: faq/prooffacehelp.php
Comment (cpeel) [reply faq/prooffacehelp.php]: In a perfect world we would do option 1. However, I have no confidence that the metadata code actually works and I don't think it's worthwhile to spend time figuring out if it does just to do that option.

My suggestion is that we update `md_phase2.php` to set the initial zoom (`$iWidth`) to the default values for those fields and be done with it.

Comment (srjfoo) [reply faq/prooffacehelp.php]: My impression was that the `md_phase2`-related code was incomplete, and certainly hasn't been maintained over the years. -- I like @cpeel's suggestion.

Comment (70ray) [reply faq/prooffacehelp.php]: Done this.
Should we also remove the database entries?

Comment (chrismiceli) [reply faq/prooffacehelp.php]: I would think it would be nice to clear up database, but that's my not very strong opinion.

File: tools/proofers/image_block_enh.inc
Comment (srjfoo) [inline tools/proofers/image_block_enh.inc:35]: This is very picky, and I don't know that I have a strong opinion one way or the other, but is using "frame" in the id (here and for `#newimageframe`) appropriate since we're moving away from coding using frames?

**Review by srjfoo — COMMENTED** [review]
For some reason, I don't seem to be able to quote some of the comments, so, constructing the quotes myself:

> @70ray: Should we also remove the database entries?

> @chrismiceli: I would think it would be nice to clear up database, but that's my not very strong opinion.

In https://github.com/DistributedProofreaders/dproofreaders/pull/651#discussion_r679546329 @cpeel said

> We do need to pull out the image zoom from userprefs, the initial DB schema, and a DB migration script as part of this. I can help with that if we want to do merge this into a temporary branch instead of master.

If that's the database cleanup you mean, then yes. (If there's another db cleanup you mean, I'm confused -- can you clarify?)

File: faq/prooffacehelp.php
Comment (cpeel) [reply faq/prooffacehelp.php]: Yes, we need to update `db_schema.sql` and have an upgrade script that drops the columns from the DB.

File: tools/proofers/image_block_enh.inc
Comment (70ray) [reply tools/proofers/image_block_enh.inc:35]: You are right. These aren't frames but divs.

File: faq/prooffacehelp.php
Comment (70ray) [reply faq/prooffacehelp.php]: I've updated `db_schema.sql` and made an upgrade script by modifying existing models.

File: SETUP/upgrade/16/20210802_clean_user_profiles.php
Comment (cpeel) [inline SETUP/upgrade/16/20210802_clean_user_profiles.php:1]: Please rename this script `20210802_alter_user_profiles.php` since we're altering the table and not removing rows from it. The script's contents look good to me.

**Review by srjfoo — APPROVED** [review]
The testing I've done looks good (I leave the fine-tuning of the code to y'all).

Do we have enough time before the next PROD rollout to get enough public testing done before rolling it out to PROD, or do we want to give it a good long time for testing and wait until September?

**Comment by cpeel** [comment]
> Do we have enough time before the next PROD rollout to get enough public testing done before rolling it out to PROD, or do we want to give it a good long time for testing and wait until September?

I would strongly suggest we do not try to rush this in the next week and rather get it in immediately after and wait until September's release to give lots of testing / complaining time.

**Comment by srjfoo** [comment]
> > Do we have enough time before the next PROD rollout to get enough public testing done before rolling it out to PROD, or do we want to give it a good long time for testing and wait until September?
> 
> I would strongly suggest we do not try to rush this in the next week and rather get it in immediately after and wait until September's release to give lots of testing / complaining time.

That's my strong preference, too, thanks.

File: SETUP/upgrade/16/20210802_clean_user_profiles.php
Comment (70ray) [reply SETUP/upgrade/16/20210802_clean_user_profiles.php]: Done this.

**Comment by cpeel** [comment]
It looks like there are still two references to these variables in `faq/pophelp/prefs/prefs_pophelp.inc` we need to remove.

**Comment by 70ray** [comment]
> It looks like there are still two references to these variables in `faq/pophelp/prefs/prefs_pophelp.inc` we need to remove.

Thanks, I've removed them now.

**Comment by cpeel** [comment]
This looks good to me. I will wait until after the August deployment to PROD on Tuesday before merging this in. That way we can give people a month to play with it on TEST before we roll it out.

**Pull request merged**
